server/datastore/mysql/sessions_test.go

```diff
@@ -66,4 +70,41 @@ func testSessionsGetters(t *testing.T, ds *Datastore) {
 	require.NotEqual(t, prevAccessedAt, sessions[0].AccessedAt)
 
 	require.NoError(t, ds.DestroyAllSessionsForUser(context.Background(), user.ID))
+
+	// session for a non-existing user
+	newSession, err = ds.NewSession(context.Background(), &fleet.Session{UserID: user.ID + 1, Key: "someotherkey"})
+	require.NoError(t, err)
+
+	gotByKey, err = ds.SessionByKey(context.Background(), newSession.Key)
+	require.NoError(t, err)
+	assert.Equal(t, newSession.ID, gotByKey.ID)
+	require.Nil(t, gotByKey.APIOnly)
+
+	gotByID, err = ds.SessionByID(context.Background(), newSession.ID)
+	require.NoError(t, err)
+	assert.Equal(t, newSession.ID, gotByKey.ID)
+	require.Nil(t, gotByKey.APIOnly)
+
+	apiUser, err := ds.NewUser(context.Background(), &fleet.User{
+		Password:   []byte("supersecret"),
+		GlobalRole: ptr.String(fleet.RoleObserver),
+		APIOnly:    true,
+	})
+	require.NoError(t, err)
+
+	// session for an api user
+	apiSession, err := ds.NewSession(context.Background(), &fleet.Session{UserID: apiUser.ID, Key: "someapikey"})
+	require.NoError(t, err)
+
+	gotByKey, err = ds.SessionByKey(context.Background(), apiSession.Key)
+	require.NoError(t, err)
+	assert.Equal(t, apiSession.ID, gotByKey.ID)
+	require.NotNil(t, gotByKey.APIOnly)
+	assert.True(t, *gotByKey.APIOnly)
+
+	gotByID, err = ds.SessionByID(context.Background(), apiSession.ID)
+	require.NoError(t, err)
+	assert.Equal(t, apiSession.ID, gotByKey.ID)
+	require.NotNil(t, gotByKey.APIOnly)
```

Comment:
When seeing a mix of `assert` and `require` like this for a nil check, I assume it's to avoid panics in tests. This is a bit of a nitpick, but you can also write this as

```
if assert.NotNil(t. gotByKey.APIOnly) {
    assert.True(t, *gotByKey.APIOnly)
}
```

Many functions in `assert` return a boolean to indicate if an assertion was successful for this purpose.

Reply:
Yeah I know, I've used it in the past but had PR comments to use the require/assert mix instead, seems to be the preferred approach here.